scripts/popup/GUIPopup.gd

```diff
@@ -3,18 +3,31 @@
 class_name GUIPopup
 extends Panel
 
-@onready var label: Label = $Label
+@onready var title: Label = $Title
+@onready var content: RichTextLabel = $Content
 @onready var yes: Button = $YesButton
 @onready var no: Button = $NoButton
 
 func set_visibility(visibility: bool) -> GUIPopup:
 	visible = visibility
 	return self
 
-func set_text(text: String) -> GUIPopup:
-	label.text = text
+func set_title(text: String) -> GUIPopup:
+	title.text = text
 	return self
-	
+
+func set_content(text: String) -> GUIPopup:
+	content.text = text
+	return self
+
+func set_yes_text(text: String) -> GUIPopup:
+	yes.text = text
+	return self
+
+func set_no_text(text: String) -> GUIPopup:
+	no.text = text
+	return self
+
```

Comment:
Separated the title text from the body text (perhaps we can rename it body_text, actually). And added text setters for the yes & no buttons.